azure-pipelines.docker.yml

```diff
@@ -0,0 +1,140 @@
+        gccVersion: '4.9'
+        installMethod: 'conda'
+        installYask: 'false'
+        imageBaseName: 'devito'
+      conda-gcc5:
+        gccVersion: '5'
+        installMethod: 'conda'
+        installYask: 'false'
+        imageBaseName: 'devito'
+      conda-gcc7:
+        gccVersion: '7'
+        installMethod: 'conda'
+        installYask: 'false'
+        imageBaseName: 'devito'
+      yask-conda-gcc7:
+        gccVersion: '7'
+        installMethod: 'conda'
+        installYask: 'true'
+        imageBaseName: 'devito-yask'
+      conda-gcc8:
+        gccVersion: '8'
+        installMethod: 'conda'
+        installYask: 'false'
+        imageBaseName: 'devito'
+
+  variables:
+     imageName: $(imageBaseName)-$(installMethod)-gcc$(gccVersion)
+     dockerFile: Dockerfile.pipelines
+     
+  steps:
+  - script: |
+      docker build -f $(dockerFile) -t $(dockerId)/$(imageName):$(Build.BuildId) --build-arg gccvers=$(gccVersion) --build-arg installMethod=$(installMethod) --build-arg yask=$(installYask) .
+      docker tag $(dockerId)/$(imageName):$(Build.BuildId) $(dockerId)/$(imageName):latest
+      docker login -u $(dockerId) -p $(dockerPassword)
+      docker push $(dockerId)/$(imageName):$(Build.BuildId)
+      docker push $(dockerId)/$(imageName):latest
+
+
+- job: Run_Tests
+  displayName: 'Test using'
+  dependsOn: Container_Build
+  condition: succeeded('Container_Build')
+  pool:
+    vmImage: 'Ubuntu 16.04'
+  strategy:
+    matrix:
+      pip-gcc49:
+        containerName: devito-pip-gcc49
+        testMethod: 'pip'
+      openmp-conda-gcc49:
+        containerName: devito-pip-gcc49
+        testMethod: 'conda'
+        useOpenMP: 'true'
+      conda-gcc5: 
+        containerName: devito-conda-gcc5
+        testMethod: 'conda'
+      yask-conda-gcc7:
+        containerName: devito-yask-conda-gcc7
+        testMethod: 'conda'
+        devitoBackend: 'yask'
+      ops-conda-gcc7:
+        containerName: devito-conda-gcc7
+        testMethod: 'conda'
+        devitoBackend: 'OPS'
+      conda-gcc8: 
+        containerName: devito-conda-gcc8
+        testMethod: 'conda'
+
+  container: $[ variables['containerName'] ]
+  timeoutInMinutes: 180
+  steps:
+  - bash: | 
+      export HOME=/devito
+  
+      if [ -v devitoBackend} ] ; then
+        export DEVITO_BACKEND=$(devitoBackend) ;
+        echo "Setting DEVITO_BACKEND=${DEVITO_BACKEND}" ;
+      fi
+
+      if [ -v useOpenMP ] ; then
+        export DEVITO_OPENMP=1 ;
+        export OMP_NUM_THREADS=2 ;
+        echo "Setting DEVITO_OPENMP=${DEVITO_OPENMP} and OMP_NUM_THREADS=${OMP_NUM_THREADS}"
+      fi
+
+      if [ $(testMethod) == 'pip' ] ; then
+        export PYTHONPATH=${PYTHONPATH}:/devito/lib/python3.6/site-packages
+        python setup.py test ; 
+      elif [ $(testMethod) == 'conda' ] ; then
+        source activate devito ;
+        py.test --cov devito tests/ ;
+        python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5 ;
```

Comment:
when the devito backend is either `yask` or `ops`, this line up until line 133 (included) shouldn't actually be run/tested

Reply:
Thanks - I'll update accordingly.